src/SuperOffice/Provider.php

```diff
@@ -105,4 +101,28 @@ public static function additionalConfigKeys(): array
             'customer_id',
         ];
     }
+
+    private function getBaseApiUrl(): string
+    {
+        $cache_time = 60 * 60 * 8; // 8 hours
+        $environment = $this->getConfig('environment', 'sod');
+        $customer_id = $this->getConfig('customer_id');
+
+        return cache()->remember('superoffice-base-url', $cache_time, function () use ($environment, $customer_id) {
+            $url = sprintf(
+                'https://%s.superoffice.com/api/state/%s',
+                $environment,
+                $customer_id
+            );
+
+            $response = $this->getHttpClient()->get($url);
+            $api_url = json_decode((string) $response->getBody(), true)['Api'];
```

Comment:
Do we need a null coalesce here? I doubt the check below will get hit because if this prop is missing then it will error here.